engine/options.go

```diff
@@ -372,6 +372,22 @@ func WithRetrievalAddrs(addrs ...string) Option {
 	}
 }
 
+// WithRetrievalPeerID sets the peerID of where to get the content corresponding to an indexing advertisement.
+// If unspecified, the libp2p host peer ID are used.
+// See: WithHost
+func WithRetrievalPeerID(peerID string) Option {
```

Comment:
This is bound to provider libp2p host, since the identity is needed for signing of advertisements. 

Have you considered setting the engine host option instead?

Reply:
You mean just using `WithHost` and set the full retrieval host? Wouldn't that cause problems?

Reply:
It shouldn't. The identity of a provider is tied to what it uses[ to sign advertisements ](https://github.com/ipni/index-provider/blob/main/engine/options.go#L55-L59) and where chain is retrievable from. 
The reason for `RetrievalAddrs` option existing is that a libp2p node may have any public addr through DNS for example, that may not directly be known by it. This is similar to setting public hostname in a traditional HTTP server.

But the peer ID (or identity by extension) must be known to the host itself and must match the ID other peers expect to encounter when fetching ad chain from it.

Reply:
For a bite more ctx, we have this PR deployed and it generates the following expected advertisements:

```
CID:          baguqeera7miiexnupzmlrusq2s5u2xd4ngqrsqormpjdtl6hkq73w2xqtkca
...
Signature: ✅ valid
Signed by: advertisement publisher
```

Reply:
@EnchanterIO What other options are set in the engine that is deployed?  I suspect the reason that works is that the peer ID set via option introduced by this PR matches the host identity?

Reply:
It does not match the identity, contrary, the problem we try to address with this PR is that if `retrievalAddr` is not empty, the host id (identity of index provider) is inserted **into adv** - but we want our separate p2p-server bitswap provider peer id coming from this option to be inserted into adv instead.

Reply:
Got it thank you

In IPNI terminology they are referred to as "Provider" and "Publisher". Provider provides the actual downloadable content , and Publisher tells the network of what is available for download from where i.e. expose the advertisement chain.

IIUC what you'd like to be able to do is to use the engine as a publisher for providers that may reside elsewhere. Right? 

Related note: I recommend looking at Extended Providers in IPNI spec.

Reply:
> IIUC what you'd like to be able to do is to use the engine as a publisher for providers that may reside elsewhere. Right?
That is correct.